common/src/main/java/io/druid/common/config/Log4jShutdown.java

```diff
@@ -95,21 +94,19 @@ public void start()
   @Override
   public void stop()
   {
-    if (callbacksRun.get()) {
-      return;
-    }
-    if (!state.compareAndSet(State.STARTED, State.STOPPED)) {
+    if (!state.compareAndSet(State.STARTED, State.STOPPING)) {
```

Comment:
Ran into a spark bug a few days ago where a shutdown hook firing during a "clean" shutdown causes problems. This method actually needs synchronized or else a shutdown hook fired from a SIGTERM won't wait for runCallbacks() to finish